fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -490,7 +613,14 @@ def initialize(user = nil, password = nil, use_shell_script = false, provider_sh
 
       @jwt = jwt
 
-      @transporter_executor = use_shell_script ? ShellScriptTransporterExecutor.new : JavaTransporterExecutor.new
+      if !use_shell_script && upload && !Helper.user_defined_itms_path? && Helper.mac? && Helper.xcode_at_least?(14)
```

Comment:
Can you place these conditions on another instance method? lile `should_use_altool?`

Reply:
done!